### ISSUE TITLE
vllm integration fails in batch model

### ISSUE THREAD

**Issue by amirarsalan90** [outlines-dev/outlines#650]
### Describe the issue as clearly as possible:

Thanks for your great work. Outline integration with vllm works fine for a small batch, but it fails for bigger batches. When the batch size is big, it seems the llm generation finishes early. In my example below, if I do the batch inference for a batch of 10 for example, the model works correctly, but increasing batch size (in my example a batch of 50), results in not properly finishing generation for some. 

### Steps/code to reproduce the bug:

```python
import vllm
import vllm.model_executor.layers.sampler as sampler
from pydantic import BaseModel
from typing import List
import torch
import pandas as pd
from outlines.serve.vllm import JSONLogitsProcessor, _patched_apply_logits_processors

# Patch the _apply_logits_processors so it is compatible with `JSONLogitsProcessor`
sampler._apply_logits_processors = _patched_apply_logits_processors

df = pd.read_csv("sample_file.csv")

class ConceptsList(BaseModel):
    concepts: List[str]

BASE_MODEL = "mistralai/Mistral-7B-Instruct-v0.2"
llm = vllm.LLM(model=BASE_MODEL, tensor_parallel_size=1, dtype=torch.float16, max_model_len=23000)

logits_processor = JSONLogitsProcessor(ConceptsList, llm.llm_engine)

full_prompts = df['text'].tolist()

batch_results = llm.generate(
    full_prompts[:50],
    sampling_params=vllm.SamplingParams(
        max_tokens=100, logits_processors=[logits_processor]
    ),
)
```

### Expected result:

```shell
'{"contepts":["Youtube Video","Sale","Vlog", "Lifestyle change"]}'
```

### Error message:

```shell
'{"contepts":["Youtube Video","Sale","Vlog",'
```

### Outlines/Python version information:

Version information
<details>
```
outlines==0.0.28
vllm==0.3.0
python 3.10.13
```
</details>

**Comment by lapp0** [comment]
Unfortunately I'm not able to reproduce with

```
import json
import vllm
import vllm.model_executor.layers.sampler as sampler
from pydantic import BaseModel
from typing import List
import torch
import pandas as pd
from outlines.serve.vllm import JSONLogitsProcessor, _patched_apply_logits_processors

# Patch the _apply_logits_processors so it is compatible with `JSONLogitsProcessor`
sampler._apply_logits_processors = _patched_apply_logits_processors

class ConceptsList(BaseModel):
    concepts: List[str]

BASE_MODEL = "mistralai/Mistral-7B-Instruct-v0.2"
llm = vllm.LLM(model=BASE_MODEL, tensor_parallel_size=1, dtype=torch.float16, max_model_len=23000)

logits_processor = JSONLogitsProcessor(ConceptsList, llm.llm_engine)

full_prompts = [
    f"Provide me a list of {i} strings with key 'concepts'"
    for i in range(20)
]

batch_results = llm.generate(
    full_prompts,
    sampling_params=vllm.SamplingParams(
        max_tokens=5000, logits_processors=[logits_processor]
    ),
)


for result in batch_results:
    for output in result.outputs:
            json.loads(output.text)
```

All 20 outputs were valid json.

Could you try again, setting `max_tokens` to a larger number? Perhaps generation is halting due to the token limit.

**Comment by lapp0** [comment]
Glad to hear you got it working!

And yes, vLLM preallocates space for activations based on your `max_model_len`, so you should expect substantially more memory use at 23,000. Additionally, you probably don't want to have Mistral run for more than 8k tokens per https://github.com/tomaarsen/attention_sinks

![mistral loss of coherence](https://private-user-images.githubusercontent.com/37621491/272347418-3a4c5634-cc1b-42d1-a35a-afb376a4f970.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc3ODA2NzgsIm5iZiI6MTcwNzc4MDM3OCwicGF0aCI6Ii8zNzYyMTQ5MS8yNzIzNDc0MTgtM2E0YzU2MzQtY2MxYi00MmQxLWEzNWEtYWZiMzc2YTRmOTcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEyVDIzMjYxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjZGM0ZDFmYWY4ZGJkZTdhYWQ5NWRlYzlmMTY4YWYzZDY5OTNlYzY2NmQzNTA1ZTUyMWUzZTA0YTIyNjU1NDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gGQzQBGd6C-sj8N7xYxcuIs2ODD4QwPd155TN4I3eco)

Outlines has an issue open to alleviate this problem https://github.com/outlines-dev/outlines/issues/511

**Comment by amirarsalan90** [comment]
Thanks for your help and the link to attention sink, yes I'll decrease max_model_len to 8000. Closing this now
